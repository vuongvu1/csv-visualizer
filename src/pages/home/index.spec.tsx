import React from "react";
import { render, cleanup } from "@testing-library/react";
import Home from "./index";

describe("Home", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
