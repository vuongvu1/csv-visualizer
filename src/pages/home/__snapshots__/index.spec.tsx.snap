// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
.c0 {
  width: 100%;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  background-color: #EFEFEF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 100%;
  max-width: 930px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: #EFEFEF;
  font-weight: normal;
  font-size: 16px;
  color: #3F7EF0;
  font-weight: bold;
  font-size: 32px;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 18px;
  margin-right: 0;
  padding: 12px;
  background-color: #FFFFFF;
  box-shadow: 0px 1px 0px rgba(165,165,168,0.2);
  border-radius: 12px;
}

.c7 {
  display: none;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 12px;
  border: 1px dashed #63AEF7;
  color: #707070;
  cursor: pointer;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
  text-align: center;
  padding: 12px;
}

.c6:hover {
  background-color: #d9eafb;
}

.c8 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 18px;
  padding: 12px;
  background-color: #FFFFFF;
  box-shadow: 0px 1px 0px rgba(165,165,168,0.2);
  border-radius: 12px;
}

@media (max-width:768px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c5 {
    margin-right: 18px;
    margin-bottom: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
        type="h1"
      >
        CSV Visualizer
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <label
            class="c6"
            for="file-id"
          >
            Click to upload
            <br />
            (accept only .csv file)
          </label>
          <input
            accept=".csv"
            class="c7"
            id="file-id"
            type="file"
          />
        </div>
        <div
          class="c8"
        >
          No data to display.
        </div>
      </div>
    </div>
  </div>
</div>
`;
